src/ngraph/op/one_hot.cpp

```diff
@@ -162,7 +162,32 @@ void op::v1::OneHot::validate_and_infer_types()
         }
         m_axis =
             ngraph::normalize_axis(this, m_axis, indices_rank + 1, -indices_rank - 1, indices_rank);
-        int64_t depth_val = as_type_ptr<op::Constant>(depth)->get_vector<int64_t>()[0];
+
+        auto depth_element_type = depth->get_output_element_type(0);
+        NODE_VALIDATION_CHECK(this,
+                              depth_element_type == element::i8 ||
+                                  depth_element_type == element::i32 ||
+                                  depth_element_type == element::i64,
+                              "'depth' input element type must be i8, i32 or i64 (got ",
+                              depth_element_type,
+                              ").");
+
+        NODE_VALIDATION_CHECK(this,
+                              is_scalar(depth->get_shape()),
```

Comment:
The old solution allows us to threat one element vector as a scalar.
It can be worth to keep this functionality.
```suggestion
                              shape_size(depth->get_shape()) == 1),
```

Reply:
They aren't scalars.